### ISSUE TITLE
Question of Lab3-1

### ISSUE THREAD

**Issue by Puff-Wen** [2017-fall-DL-training-program/VAE-GAN-and-VAE-GAN#1]
Hi TA,

1. In Hyper-parameters, what's the meaning of beta:6 and latent code size:10 ?
2. What does "change latent codes from -2 to 2" means described in second part of disentanglement ?

Thanks

**Comment by hui-po-wang** [comment]
Hi,

For your first question, beta stands for a constant which enhances the KL divergence in the original loss function of VAE and force the posterior q(z|x) to be more compatible with the prior (in our case, 10-D standard normal distribution). Please refer to the pdf file and check the loss function in beta-VAE section. Latent code size means the dimension of latent codes. During the training and the testing, the coder will take a batch of images as inputs, and "encode" them to be a batch of "mean" and "variance".  In your assignment, the size (dim.) of your latent codes should be (batch_size, latent code size). Note that the latent code should be drawn from a 10-D Gaussian of which the mean and variance are produced by the encoder.

For the second question, the assignment asks you to change the latent code along one axis each time. That is, for example, Let's assume that we have a set of latent codes, which is encoded from input data, with the size (batch_size=1, dim=2) and the value (0, 1). When we try to draw the final result pirture, we first change the first axis of the latent codes (the dim. of 0) from 2 to -2 uniformly, so the first element in the first raw of the picture would be (-2, 1), the second element in the first raw would be (0, 1), and the third element in the first raw would be (2, 1) (for simplicity, I only take three points from the interval between -2 and 2, and you should use more points (e.q. 10) to show that you actually learn the disentanglement ability). With the similar idea, the elements in the second raw would be (0, -2),  (0, 0), and (0, 2) (still, I only draw three points from the interval).

Thanks

**Comment by yyrkoon27** [comment]
Dear TA,

In your example, can we make the second row 
`(0, -1), (0, 1), (0, 3),`
instead of
`(0, -2), (0, 0), (0, 2)?`

That means (original_code +- 2). In this way, the center element in each row will be the original reconstructed image.

Thank you very much!

**Comment by hui-po-wang** [comment]
Hi @yyrkoon27,

Sure, you can do it. The reason why I use 0 as the origin is that I "believe" that after the training procedure, our latent codes should be compatible with standard normal distributions which have "zero" mean and unit variance. However, the latent codes could not exactly match the prior, so you can do it as you mentioned above.

Thanks

**Comment by yyrkoon27** [comment]
Dear TA,

Sure, cool. Thank you very much!
